controllers/helmchart_controller.go

```diff
@@ -512,13 +512,10 @@ func (r *HelmChartReconciler) reconcileFromTarballArtifact(ctx context.Context,
 		defer src.Close()
 
 		var valuesData []byte
-		if _, err := src.Read(valuesData); err == nil {
-			isValuesFileOverriden, err = helm.OverwriteChartDefaultValues(helmChart, valuesData)
-		}
-
-		if err != nil {
+		if _, err := src.Read(valuesData); err != nil {
 			return sourcev1.HelmChartNotReady(chart, sourcev1.ChartPackageFailedReason, err.Error()), err
 		}
+		isValuesFileOverriden, err = helm.OverwriteChartDefaultValues(helmChart, valuesData)
```

Comment:
As discussed in private, but noted here for the public record. We should return this error.